### ISSUE TITLE
JustWatch Not Found error since 2023/10/12

### ISSUE THREAD

**Issue by dduives** [haijeploeg/excludarr#92]
**Describe the bug**
Has always been working fine but has been throwing the error **excludarr.modules.justwatch.exceptions.JustWatchNotFound** since 2023/10/12

Before that it sometimes worked but occasionally has thrown this error:
**Max retries exceeded with url: /content/locales/state** 

After that it has been throwing the JustWatchNotFound error for the last 8 days

**Versions**
1. OS Version: Docker on Synology
2. Excludarr version: 1.0.7
3. Python version: 3.10

**Log:**
[excludarr.csv](https://github.com/haijeploeg/excludarr/files/13054408/excludarr.csv)

**Comment by LucaVignati** [comment]
I have the same error, I am on Unraid 6.12.4

**Comment by reaper** [comment]
+1

**Comment by Pauwelz** [comment]
Probably because the JustWatch API address defined [justwatch.py @ line 13](https://github.com/haijeploeg/excludarr/blob/6aacf87a0e812bec9158301831059bb724fb5347/excludarr/modules/justwatch/justwatch.py#L13C26-L13C60) no longer exists.

Seems like it was replaced with a GraphQL endpoint located at [https://apis.justwatch.com/graphql](https://apis.justwatch.com/graphql).

Sadly I'm not fluent enough in Python to fix this. Maybe someone else can pick this up.

**Comment by haijeploeg** [comment]
Yep, they have changed the API to a graphql endpoint. Im not sure how to quickfix this as the whole API client needs to be rewritten. Sadly I don't have any spare time anymore to maintain this tool... Maybe in the future I will look into this.

**Comment by keldian** [comment]
Could you not leverage https://github.com/Electronic-Mango/simple-justwatch-python-api for this?

**Comment by TheArturthur** [comment]
> Could you not leverage https://github.com/Electronic-Mango/simple-justwatch-python-api for this?

Apparently no. In terms of licensing, as that repo is under GPLv3, this one must be under the same license (currently is using MIT).

I've forked this project to try and come up with a solution using GraphQL requests. I don't have much time, but will try my best.

**Comment by jelias** [comment]
@TheArturthur curious what the status of your GQL implementation is? Was looking through your code and it seemed decently far along...

Were you able to get a working implementation?

**Comment by TheArturthur** [comment]
No, I've been quite busy to sit down with it. I've managed to try requests that should be made to the API, though. Hopefully, my schedule will allow me to spend some time with it...

**Comment by g-nogueira** [comment]
I really want to use this project also. After I'm done with some changes I would personally like on botdarr, I will give a try on this one.

**Comment by scott-es** [comment]
I was looking into this project rather than try to create something from scratch, but it appears it's unmaintained.  @haijeploeg are you looking to transfer ownership of this project to continue to evolve it?  Is there a better place (e.g. some discord server) to discuss the future of a solution for the use cases solved by Excludarr?  

I may be able to make the updates to work with GQL for justwatch, but no PRs on this repo have been merged since '22, worst case we end up with a fork with a bunch of needed updates ...

**Comment by TheArturthur** [comment]
Is it possible/legal for us to use any of the created forks for this, even without his response? Or even a new one inside an org?
Obviously, giving him credit for the original idea and development, but being able to improve it.

**Comment by scott-es** [comment]
> Is it possible/legal for us to use any of the created forks for this, even without his response? Or even a new one inside an org? Obviously, giving him credit for the original idea and development, but being able to improve it.

Yes, it's MIT license

**Comment by TheArturthur** [comment]
Great, I just created a free org, imported my fork, and made you all owners. Feel free to edit the org config as you think best!

https://github.com/excludarr

**Comment by scott-es** [comment]
> Great, I just created a free org, imported my fork, and made you all owners. Feel free to edit the org config as you think best!
> 
> https://github.com/excludarr

Great!  thank you for doing that! accepted the invite.  I will work on submitting an update to fix this issue, as soon as I have something of decent progress I'll commit and raise a WIP PR.  Will continue discussion then on other new repository. Too bad there is not a way (AFAIK) to "take over" abandoned projects but at least the new org will have multiple owners and maintainers hopefully.

**Comment by scott-es** [comment]
let's continue the discussion about fixing this in the corresponding [issue](https://github.com/excludarr/excludarr/issues/2) in the new fork

**Comment by barsa2000** [comment]
if someone is interested I'm continuing the work from this repository because excludarr/excludarr was unresponsive when i added my pull requests :)

There still is some cleanup work to do, but it's working with my libraries. 

check it out at:
https://github.com/barsa2000/excludarr

**Comment by jloh** [comment]
@barsa2000 whats the best way to run your fork, is there a docker container or something? Would be fantastic if the readme could be updated, I'd love to try it out.

**Comment by barsa2000** [comment]
ATM you can either build the docker yourself (the dockerfile was working last time I checked) or run the code with poetry, tomorrow I'll try to publish it on dockerhub so it's gonna be easier to use. 

The readme update is a todo, I just prioritized getting it to a working state, I'll update it tomorrow :)

If you find any problems just open an issue and I'll check it :)

**Comment by brokenwu** [comment]
So now that barsa2000s fork also seems like it's dead, any other way to run this?

**Comment by barsa2000** [comment]
https://github.com/excludarr/excludarr
